src/arel/_middleware.py

```diff
@@ -0,0 +1,218 @@
+            self._script_template = _Template(SCRIPT_TEMPLATE_PATH.read_text())
+
+        scheme = {
+            "http": "ws",
+            "https": "wss",
+        }[scope["scheme"]]
+        # NOTE: 'server' is optionaly in the ASGI spec. In practice,
+        # we assume it is effectively provided by server implementations.
+        host, port = scope["server"]
+        path = self._ws_path
+
+        url = f"{scheme}://{host}:{port}{path}"
+
+        js = self._script_template.substitute(
+            {"url": url, "reconnect_interval": self._reconnect_interval}
+        )
+
+        return f'<script type="text/javascript">{js}</script>'.encode("utf-8")
+
+    async def _handle_http(self, scope: Scope, receive: Receive, send: Send) -> None:
+        script = self._get_script(scope)
+        inject_script = False
+
+        async def wrapped_send(message: Message) -> None:
+            nonlocal inject_script
+
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(raw=message["headers"])
+
+                if headers.get("content-type", "").partition(";")[0] != "text/html":
+                    # This is not HTML.
+                    await send(message)
+                    return
+
+                if headers.get("transfer-encoding") == "chunked":
+                    # Ignore streaming responses.
+                    await send(message)
+                    return
+
+                if headers.get("content-encoding"):
+                    msg = textwrap.dedent(
+                        f"""
+                        Cannot inject reload script into response encoded as {headers['content-encoding']!r}.
+
+                        HINT: 'HotReloadMiddleware' must be placed inside any content-encoding middleware, such as:
+
+                        middleware = [
+                            Middleware(GZipMiddleware),
+                            Middleware(HotReloadMiddleware, ...),
+                        ]
+                        """  # noqa: E501
+                    )
+                    warnings.warn(msg)
+
+                    await send(message)
+                    return
+
+                inject_script = True
+
+                if "content-length" in headers:
+                    new_length = int(headers["content-length"]) + len(script)
+                    headers["content-length"] = str(new_length)
+
+                await send(message)
+
+            else:
+                assert message["type"] == "http.response.body"
+
+                if not inject_script:
+                    await send(message)
+                    return
+
+                if message.get("more_body", False):
+                    raise RuntimeError("Unexpected streaming response")
+
+                body: bytes = message["body"]
+
+                try:
+                    start = body.index(b"</body>")
+                except ValueError:
+                    await send(message)
```

Comment:
Content length is here updated with the script, but script is not attached, creating a RuntimeError in Starlette (`RuntimeError: Response content shorter than content length`).

This check should be moved to the conditional above: `if message["type"] == "http.response.start": ...`, e.g. like this on L185:
```
if b"</body>" not in message["body"]:
      await send(message)
      return
```